staging/src/k8s.io/apimachinery/pkg/labels/selector.go

```diff
@@ -943,7 +943,7 @@ func ValidatedSelectorFromSet(ls Set) (Selector, error) {
 // A nil and empty Sets are considered equivalent to Everything().
 // It assumes that Set is already validated and doesn't do any validation.
 func SelectorFromValidatedSet(ls Set) Selector {
-	if ls == nil || len(ls) == 0 {
+	if len(ls) == 0 {
```

Comment:
`Set` is essentially a map, and the default value of a map is `nil`. This check is valid imo :)

Reply:
Edit: although a `nil` map has Len of 0

Reply:
yeah, even nil ones, which have a length of zero. So I think It is not necessary to check nil.

Reply:
@MadhavJivrajani Thanks for your review, should I revert here?